sw/device/boot_rom/boot_rom.c

```diff
@@ -15,21 +15,13 @@
 #include "sw/device/lib/spi_device.h"
 #include "sw/device/lib/uart.h"
 
-static noreturn void try_launch(void) {
-  asm volatile(
-      "  la   sp, _stack_start;"
-      "  tail _flash_start;"
-      // Unfortunately, we can't tell the compiler that we're about to clobber
-      // the stack pointer (which is meaningless, since there's nowhere to spill
-      // sp to). However, this doesn't matter, since this asm block will never
-      // return.
-      :
-      :
-      : /* sp, */ "memory");
-  __builtin_unreachable();
-}
+// NOTE: This symbol is generated by the linker.
+// See rom_link.ld and flash_link.ld.
+extern struct {
+  void (*entry)(void);
+} _flash_start;
```

Comment:
I don't see this symbol in the linker script.  Also, the ROM image won't have a header, it needs to depend on Ibex reset vector invocation.

Reply:
I clarified the comment to explain what comes from what. It's a bit confusing, seeing as how this is the bridge between two binaries.